### DISCUSSION TITLE
KSQLDB-Server got error when creating stream

### DISCUSSION THREAD

**Discussion by truongbk24** [Apicurio/apicurio-registry#2829] [Q&A]
Hello there, i'm trying to setup the KSQLDB to create the stream pipeline from different topic. I used Apicurio as a schema registry and the Keycloak for authentication, i created the registry-api client id for the api and apicurio-registry client id to work with ui Web console and api for kafka connector. Everything is working fine, but when i tried to create the stream in KSQL server with the information:

ksql.schema.registry.url=http://registry:8080
bootstrap.servers=kafka:9092
ksql.logging.processing.topic.auto.create=true
ksql.schema.registry.basic.auth.user.info=test:t123456
listeners=http://0.0.0.0:8088
ksql.schema.registry.basic.auth.credentials.source=USER_INFO
ksql.logging.processing.stream.auto.create=true

I also try the url: http://registry:8080/apis/ccompat/v6 but it did not working. Is there a case to provide the client id and secret to authenticate via keycloak or did i do something wrong. Thank you for your support

**Comment by EricWittmann** [comment]
What sort of error do you get from ksqldb?  If you have BASIC auth enabled in the Apicurio Registry server, then you shouldn't need to care about the client id and secret in the ksqldb config.

I would recommend trying to make a REST API call to apicurio registry using BASIC auth credentials via something like `curl` or Postman to make sure that works first.  If that works ok, then we can dig more into ksqldb maybe.
